src/UbuntuToolkit/ucunits.cpp

```diff
@@ -322,8 +326,16 @@ void UCUnits::windowPropertyChanged(QPlatformWindow *window, const QString &prop
     if (!ok || scale <= 0) {
         return;
     }
+
+    // Update our pixel ratio to make sure it has not changed
+    m_devicePixelRatio = qGuiApp->devicePixelRatio();
+
     // choose integral grid unit value closest to requested scale
-    setGridUnit(qCeil(scale * DEFAULT_GRID_UNIT_PX) * m_devicePixelRatio);
+    // If qt hidpi scaling is enabled, use that as scaling
+    if (QHighDpiScaling::isActive())
+      setGridUnit(qCeil(m_devicePixelRatio * DEFAULT_GRID_UNIT_PX));
+    else
+      setGridUnit(qCeil(scale * DEFAULT_GRID_UNIT_PX));
```

Comment:
Shouldn't we keep this line as it was before, and multiply the value by `m_devicePixelRatio`?

Reply:
Looking at http://code.qt.io/cgit/qt/qtbase.git/tree/src/gui/kernel/qhighdpiscaling.cpp?h=5.9.6#n399 it seems to return 1

Reply:
Just to play 100% safe, I'd add it back, but it's your call :-)